src/plugin/areadata/areadataplugin.js

```diff
@@ -0,0 +1,120 @@
+goog.module('plugin.areadata.AreaDataPlugin');
+goog.module.declareLegacyNamespace();
+
+
+const AbstractPlugin = goog.require('os.plugin.AbstractPlugin');
+const registry = goog.require('os.ogc.registry');
+const services = goog.require('os.ogc.services');
+const OGCService = goog.require('os.ogc.OGCService');
+const {AreaImportType} = goog.require('plugin.areadata');
+
+const OSSettings = goog.require('os.config.Settings');
+
+const Settings = OSSettings.getInstance();
+
+
+/**
+ * @type {string}
+ * @const
+ */
+const ID = 'areadata';
+
+/**
+ * @type {string}
+ * @const
+ */
+const KEYS_SETTING = 'areadata.keys';
+
+/**
+ * Maps a non-minimized os.ui.menu.MenuItemOptions object, e.g. from settings.json, onto the minimized version
+ *
+ * WARNING: To avoid turning a common object into an extern, only the primitive properties defined as of 20200812
+ * are mapped to a new os.ui.menu.MenuItemOptions object. Please add new mappings as needed.
+ *
+ * @param {Object.<string, *>|undefined|null} config
+ * @return {os.ui.menu.MenuItemOptions|undefined}
+ */
+const minify_ = function(config) {
+  if (!config) return undefined;
+
+  return /** @type {os.ui.menu.MenuItemOptions} */ ({
+    eventType: config['eventType'],
+    label: config['label'],
+    metricKey: config['metricKey'],
+    visible: config['visible'],
+    enabled: config['enabled'],
+    selected: config['selected'],
+    icons: config['icons'],
+    tooltip: config['tooltip'],
+    shortcut: config['shortcut'],
+    sort: config['sort']
+  });
+};
+
+/**
+ *
+ */
+class AreaDataPlugin extends AbstractPlugin {
+  /**
+   *
+   */
```

Comment:
Nit: our linter requires these stupid extra comment blocks as a holdover from the pre-ES6 days. This shouldn't be necessary, but if we're gonna have them, we might as well put something in them.